### PR TITLE
fix(vault): parse block-style YAML lists in frontmatter (not just inline arrays)

### PR DESCRIPTION
**Pull request by lymeswold**
## Problem

`o2b`'s frontmatter reader, `parseFrontmatterText` in `src/core/vault.ts`, only recognises **inline YAML arrays**:

```yaml
tags: [brain, brain/signal, brain/topic/foo]
```

It has no code path for **block-style (standard YAML) lists**:

```yaml
tags:
  - brain
  - brain/signal
  - brain/topic/foo
```

The `tags:` line has an empty value and the `- brain` lines are not `key: value` pairs, so the key is silently skipped. The parser returns no `tags`, and the signal validator then throws `signal missing field: tags` — even though the field is plainly present.

## When it occurs

Any time a `Brain/**/*.md` file uses block-style lists. Concrete triggers:
- **Obsidian's Properties editor** — converts inline arrays to block lists on edit.
- **Writers that emit standard YAML** (e.g. an agent recording a signal via a tool that renders block-style frontmatter).
- **Hand edits** in a normal editor.

In practice this produces a swarm of `signal-invalid` errors from `o2b brain doctor` on otherwise-valid files.

## Reproduction (before fix)

Create `Brain/inbox/sig-test.md`:

```yaml
---
kind: brain-signal
id: sig-test
created_at: 2026-07-18T00:00:00Z
tags:
  - brain
  - brain/signal
  - brain/topic/foo
topic: foo
signal: positive
agent: test
principle: test
---
```

`o2b brain doctor` reports `signal-invalid: signal missing field: tags`.

## Fix

Teach the parser to assemble dash-item blocks under an empty-value key into an array, while preserving the original empty-string behaviour for a genuine null scalar (`key:` with no following dash list, so it still round-trips as `""`). Inline-array parsing is unchanged — purely additive.

## Test coverage (tests/core/vault.test.ts)

1. Block list → array (the core regression).
2. Quoted items inside a block list parse as strings.
3. A bare `key:` *not* followed by a dash list stays an empty string (disambiguation — the tricky part).
4. Block lists and inline arrays interleaving in one file.

All existing `vault` / `signal` / `doctor` suites pass (29 in vault.test.ts; 66 across signal+doctor). Formatter hook (`oxfmt`) satisfied.

## Prior art check

No existing issue or PR covers this: searching upstream issues for `"signal missing field"` and `parseFrontmatter block` returns 0 results, and the recent frontmatter PRs (#139 write-path integrity, #132 ingestion robustness, #76 write-time governance) do not touch `parseFrontmatterText`'s block-sequence limitation.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved frontmatter parsing to properly recognize YAML block lists (e.g., `key:` followed by `- item` entries) and convert them into arrays.
  - Preserved special cases such as list items containing `/` or commas, and kept quoted/comma-containing entries intact.
  - Maintained correct behavior for empty scalar values when no list follows, including isolated empty dash lines.
  - Allowed mixing inline arrays and block lists in the same frontmatter without interference.
- **Tests**
  - Added regression coverage for the new frontmatter parsing behaviors and edge cases.

### CODE REVIEW

**Comment by solaitken** [comment]
Thank you for this fix, it is genuinely useful. Block-style lists are exactly what Obsidian's Properties editor emits, so this closes a real gap between the Obsidian-native promise and what the parser actually accepted, and the disambiguation of a bare `key:` scalar is handled correctly with tests.

One request: could you take a look at the two CodeRabbit comments when you get a chance? They point at a real edge case - a bare `-` (empty list item) currently fails the dash regex and silently drops the rest of the list. The suggested module-level `DASH_ITEM_RE` plus `?? ""` handles it and would fit the project's no-silent-drops rule. A small regression test for the empty-item case would be welcome too.

**Review by itechmeat — APPROVED** [review]

File: src/core/vault.ts [hunk 1]
```diff
@@ -81,14 +81,61 @@ export function parseFrontmatterText(text: string): readonly [FrontmatterMap, st
...
+    const line = lines[i]!.trim();
     if (!line || line.startsWith("#")) continue;
+
+    const dash = isDashItem(line);
+    if (dash && blockKey !== null) {
+      const arr = (metadata[blockKey] as string[] | undefined) ?? [];
+      arr.push(stripQuotes(dash[1]!));
+      metadata[blockKey] = arr;
+      continue;
+    }
```
[the remark this answers is not in the document]
Comment (lymeswold) [reply src/core/vault.ts]: Addressed in 496b785. The dash matcher is now module-scoped as DASH_ITEM_RE = /^-(?:\s+(.*))?$/, which matches - foo, - , and a bare - while still rejecting-foo (no whitespace → not a list item). Empty items now surface as "" via dash[1] ?? "" instead of resetting blockKey and silently dropping the rest of the list.Covered by a new regression test (empty-item block list is not dropped).

File: src/core/vault.ts [hunk 2]
```diff
@@ -81,14 +81,61 @@ export function parseFrontmatterText(text: string): readonly [FrontmatterMap, st
...
+      while (j < lines.length) {
+        const cand = lines[j]!.trim();
+        if (!cand || cand.startsWith("#")) {
+          j++;
+          continue;
+        }
+        nextMeaningful = cand;
+        break;
+      }
+      if (nextMeaningful !== null && isDashItem(nextMeaningful)) {
```
[the remark this answers is not in the document]
Comment (lymeswold) [reply src/core/vault.ts]: Done — the lookahead now calls DASH_ITEM_RE.test(nextMeaningful) instead of the old inline isDashItem, reusing the same module-scoped regex. The nextMeaningful!== null guard is preserved. Added a regression test confirming -foo (no whitespace) is not parsed as a list item.

**Comment by lymeswold** [comment]
Grateful for the review and the approval. Both CodeRabbit points are now addressed in commit 496b785:

• Empty list items no longer silently dropped. The dash matcher is now module-scoped as DASH_ITEM_RE = /^-(?:\s+(.*))?$/, which matches - foo, - , and a bare -,while still rejecting -foo (no whitespace → not a list item). Empty items surface as "" rather than resetting blockKey and aborting the rest of the list.
• Lookahead reuses the same regex (DASH_ITEM_RE.test(...)) instead of the old inline arrow function.

Added two regression tests: an empty-item block list is not dropped (["brain", "", "brain/signal"]), and a dash-prefixed string without whitespace (-foo) is correctly not treated as a list item.

All suites green: 31 in vault.test.ts, 66 across signal/doctor, plus tsc --noEmit and the formatter. Ready for re-review / merge.

Great repo ... enjoying it ... many thanks!  Chris N. (lymeswold), London, UK.

**Review by itechmeat — APPROVED** [review]

**Comment by itechmeat** [comment]
Thanks for the feedback and the kind words, Chris @lymeswold! Really glad you're enjoying the repo.

**Comment by lymeswold** [comment]
Just a heads up on CI: the validate job is currently red, but it's not from my change. The only failing test is temporalReplace logs and temporal-replace event (tests/core/brain/lifecycle/temporal-replace.test.ts:127 — expect(events.length).toBe(1), received 0).

My PR only touches src/core/vault.ts and tests/core/vault.test.ts; the failing test lives in temporal-replace.ts, which doesn't import from vault.ts. More telling: main itself is red on the same job (run 29675080196, same 4b8100ca head) with the identical failure - so it's a pre-existing breakage on main, independent of this branch (6357/6358 pass here).

I've left that test alone since it's outside scope and you own. But happy to rebase / force-push once main is green, or fold in a fix if you'd prefer it to travel with this PR.
